src/main/java/org/mqttbee/mqtt/handler/websocket/WebSocketClientProtocolHandlerOnHandshakeComplete.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 The MQTT Bee project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mqttbee.mqtt.handler.websocket;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.websocketx.WebSocketClientProtocolHandler;
+import io.netty.handler.codec.http.websocketx.WebSocketVersion;
+
+import java.net.URI;
+
+public class WebSocketClientProtocolHandlerOnHandshakeComplete extends WebSocketClientProtocolHandler {
+    private Runnable action;
+
+    public WebSocketClientProtocolHandlerOnHandshakeComplete(URI serverUri) {
+//        super(serverUri, WebSocketVersion.V13, "mqttv3.1", true, new DefaultHttpHeaders().add("Sec-WebSocket-Protocol", "mqttv3.1"), 128000);
+        super(serverUri, WebSocketVersion.V13, "mqttv3.1", true, null, 128000);
+    }
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+
+        if (ClientHandshakeStateEvent.HANDSHAKE_COMPLETE.equals(evt)) {
+            // action allows other handlers to be added to the pipeline once the handshake is complete.
+
+            ctx.channel().pipeline().addLast("ws-binaryframe-creator", new WebSocketBinaryFrameEncoder());
+            ctx.channel().pipeline().addLast("ws-binaryframe-unpacker", new WebSocketBinaryFrameDecoder());
+            action.run();
```

Comment:
Call MqttWebSocketsChannelInitializer.addMqttHandlers directly (easier to understand)
Pass MqttWebSocketsChannelInitializer as argument to the constructor

Reply:
I looked at this again. I think I actually prefer to remove these pipeline manipulations from the handler and move them to the ChannelInitializer. This way it is easier to see the pipeline mods in one place.